### DISCUSSION TITLE
dialyzer is not happy with otel_meter.hrl macros

### DISCUSSION THREAD

**Discussion by RoadRunnr** [open-telemetry/opentelemetry-erlang#707] [Q&A]
otel_meter.hrl uses some macros that all of this general form:

```
-define(counter_add(NameOrInstrument, Number, Attributes),
        case is_atom(NameOrInstrument) of 
            true -> 
                otel_counter:add(otel_ctx:get_current(), ?current_meter, NameOrInstrument, Number, Attributes); 
            false -> 
                otel_counter:add(otel_ctx:get_current(), NameOrInstrument, Number, Attributes) 
        end).
```

When those macros are use like `?counter_add('my.super.meter', 1, #{})` then dialyzer complains loudly about it with a `The pattern 'false' can never match the type 'true'` warning.

obviously dialyzer is right in this case, `is_atom()` will always be `true` and the false case can never match.

But the question is now, how to stop dialyzer complaining about this?

**Comment by tsloughter** [comment]
Argh, hm, I don't know besides using `-dialyzer_ignore`, but that'd be the wrong solution to require people do that for every use of the meter macros...

We may have to change the macros to split them up into 2.
